src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -5378,7 +5515,9 @@ public object VisitTryStatement(TryStatementAst tryStatementAst)
                 {
                     // This might be worth a strict-mode check - if there was a typo, the typo isn't discovered until
                     // the first time an exception is raised, which is rather unfortunate.
-                    catchTypesExpr = Expression.Block(dynamicCatchTypes.Append(catchTypesExpr));
+                    dynamicCatchTypes.Add(catchTypesExpr);
+
+                    catchTypesExpr = Expression.Block(dynamicCatchTypes.ToArray());
```

Comment:
Not that I can see, it changes the method overload it would call from the `params Expression[] expressions` to the `IEnumerable<Expression>` overload, but other than that... 🤷